test/unit-tests/mobile/devices-service.ts

```diff
@@ -932,4 +967,520 @@ describe("devicesService", () => {
 			assert.deepEqual(devicesService.getDeviceByDeviceOption(), androidDevice);
 		});
 	});
+
+	describe("isCompanionAppInstalledOnAllDevices", () => {
+		let $companionAppService: ICompanionAppsService;
+		let deviceIdentifiers = [iOSDevice.deviceInfo.identifier, androidDevice.deviceInfo.identifier];
+
+		beforeEach(() => {
+			$companionAppService = testInjector.resolve("companionAppsService");
+			$companionAppService.getCompanionAppIdentifier = (framework: string, platform: string): string => "companion-app";
+
+			(<any>devicesService).getDeviceByIdentifier = (identifier: string) => {
+				return identifier === androidDevice.deviceInfo.identifier ? androidDevice : iOSDevice;
+			};
+		});
+
+		it("should return correct result for all of the devices passed as argument.", () => {
+			let expectedResult = [true, true];
+			mockIsAppInstalled([iOSDevice, androidDevice], expectedResult);
+
+			let result = _.map(devicesService.isCompanionAppInstalledOnDevices(deviceIdentifiers, constants.TARGET_FRAMEWORK_IDENTIFIERS.Cordova),
+				(future: IFuture<IAppInstalledInfo>) => future.wait().isInstalled);
+
+			assert.deepEqual(result.length, deviceIdentifiers.length);
+		});
+
+		it("should return correct result when all of the devices have the companion app installed.", () => {
+			let expectedResult = [true, true];
+			mockIsAppInstalled([iOSDevice, androidDevice], expectedResult);
+
+			let result = _.map(devicesService.isCompanionAppInstalledOnDevices(deviceIdentifiers,
+				constants.TARGET_FRAMEWORK_IDENTIFIERS.Cordova), (future: IFuture<IAppInstalledInfo>) => future.wait().isInstalled);
+
+			assert.sameMembers(result, expectedResult);
```

Comment:
isn't `deepEqual` applicable here (and for the rest of the tests). The order of the results is important, but `sameMembers` does not check the order of elements.